homeassistant/components/switchbot/cover.py

```diff
@@ -102,3 +107,95 @@ def _handle_coordinator_update(self) -> None:
+    _attr_device_class = CoverDeviceClass.BLIND
+    _attr_supported_features = (
+        CoverEntityFeature.OPEN_TILT
+        | CoverEntityFeature.CLOSE_TILT
+        | CoverEntityFeature.STOP_TILT
+        | CoverEntityFeature.SET_TILT_POSITION
+    )
+    CLOSED_UP_THRESHOLD = 80
+    CLOSED_DOWN_THRESHOLD = 20
+
+    def __init__(self, coordinator: SwitchbotDataUpdateCoordinator) -> None:
+        """Initialize the Switchbot."""
+        super().__init__(coordinator)
+        self._attr_is_closed = None
+
+    async def async_added_to_hass(self) -> None:
+        """Run when entity about to be added."""
+        await super().async_added_to_hass()
+        last_state = await self.async_get_last_state()
+        if not last_state or ATTR_CURRENT_TILT_POSITION not in last_state.attributes:
+            return
+
+        self._attr_current_cover_tilt_position = last_state.attributes.get(
+            ATTR_CURRENT_TILT_POSITION
+        )
+        self._last_run_success = last_state.attributes.get("last_run_success")
+        if (_tilt := self._attr_current_cover_position) is not None:
+            self._attr_is_closed = (_tilt < self.CLOSED_DOWN_THRESHOLD) or (
+                _tilt > self.CLOSED_UP_THRESHOLD
+            )
+
+    async def async_open_cover_tilt(self, **kwargs: Any) -> None:
+        """Open the tilt."""
+
+        _LOGGER.debug("Switchbot to open blind tilt %s", self._address)
+        self._last_run_success = bool(await self._device.open())
+        self.async_write_ha_state()
+
+    async def async_close_cover_tilt(self, **kwargs: Any) -> None:
+        """Close the tilt."""
+
+        _LOGGER.debug("Switchbot to close the blind tilt %s", self._address)
+        self._last_run_success = bool(await self._device.close())
+        self.async_write_ha_state()
+
+    async def async_stop_cover_tilt(self, **kwargs: Any) -> None:
+        """Stop the moving of this device."""
+
+        _LOGGER.debug("Switchbot to stop %s", self._address)
+        self._last_run_success = bool(await self._device.stop())
+        self.async_write_ha_state()
+
+    async def async_set_cover_tilt_position(self, **kwargs: Any) -> None:
+        """Move the cover tilt to a specific position."""
+        position = kwargs.get(ATTR_TILT_POSITION)
+
+        _LOGGER.debug("Switchbot to move at %d %s", position, self._address)
+        self._last_run_success = bool(await self._device.set_position(position))
+        self.async_write_ha_state()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        _tilt = self.parsed_data["tilt"]
+        _moving_up = (
+            self.parsed_data["motionDirection"]["up"] and self.parsed_data["inMotion"]
+        )
+        _moving_down = (
+            self.parsed_data["motionDirection"]["down"] and self.parsed_data["inMotion"]
+        )
+        # NOTE: when motion is down, motion up is also set to true for some reason
```

Comment:
@BelowAverageDev 

After some debugging today I made https://github.com/Danielhiversen/pySwitchbot/pull/188

Reply:
Thanks for the fix @jesserockz. Updated pr (now way cleaner; thanks!): https://github.com/home-assistant/core/pull/86990

Reply:
Also note you can make a PR code suggestion instead of a full blown PR that runs through everything.